src/components/QueryEditor.tsx

```diff
@@ -1,41 +1,167 @@
+              },
+            ],
+          },
+          {
+            value: 'error',
+            label: 'Error',
+            children: [
+              {
+                value: 'rootCauseService',
+                label: 'Root Cause',
+                queryType: XrayQueryType.getAnalyticsRootCauseErrorService,
+              },
+              {
+                value: 'path',
+                label: 'Path',
+                queryType: XrayQueryType.getAnalyticsRootCauseErrorPath,
+              },
+              {
+                value: 'message',
+                label: 'Error Message',
+                queryType: XrayQueryType.getAnalyticsRootCauseErrorMessage,
+              },
+            ],
+          },
+          {
+            value: 'fault',
+            label: 'Fault',
+            children: [
+              {
+                value: 'rootCauseService',
+                label: 'Root Cause',
+                queryType: XrayQueryType.getAnalyticsRootCauseFaultService,
+              },
+              {
+                value: 'path',
+                label: 'Path',
+                queryType: XrayQueryType.getAnalyticsRootCauseFaultPath,
+              },
+              {
+                value: 'message',
+                label: 'Error Message',
+                queryType: XrayQueryType.getAnalyticsRootCauseFaultMessage,
+              },
+            ],
+          },
+        ],
+      },
+      {
+        value: 'user',
+        label: 'End user impact',
+        queryType: XrayQueryType.getAnalyticsUser,
+      } as QueryTypeOption,
+      {
+        value: 'url',
+        label: 'URL',
+        queryType: XrayQueryType.getAnalyticsUrl,
+      },
+      {
+        value: 'statusCode',
+        label: 'HTTP status code',
+        queryType: XrayQueryType.getAnalyticsStatusCode,
+      },
+    ],
+  },
+];
+
+function findOptionForQueryType(queryType: XrayQueryType, options: any = queryTypeOptions): QueryTypeOption[] {
+  for (const option of options) {
+    const selected: QueryTypeOption[] = [];
+    if (option.queryType === queryType) {
+      selected.push(option);
+      return selected;
+    }
+    if (option.children) {
+      const found = findOptionForQueryType(queryType, option.children);
+      if (found.length) {
+        selected.push(option, ...found);
+        return selected;
+      }
+    }
+  }
+  return [];
+}
 
 /**
  * We do some mapping of the actual queryTypes to options user can select. Mainly don't want user to choose
  * between trace list and single trace and we detect that based on query. So trace list option returns single trace
  * if query contains single traceID.
  */
-export enum QueryTypeOptions {
-  traceList = 'Trace List',
-  traceStatistics = 'Trace Statistics',
-}
+export function queryTypeToQueryTypeOptions(queryType?: XrayQueryType): QueryTypeOption[] {
+  if (!queryType || queryType === XrayQueryType.getTimeSeriesServiceStatistics) {
+    return [queryTypeOptions[1]];
```

Comment:
Would it make sense to return [traceStatisticsOption] here instead of queryTypeOptions[1]? Same for the getTrace below.

Reply:
Yes probably forgot to change it after I extracted those options.